drwmutex.go

```diff
@@ -122,11 +136,15 @@ func (dm *DRWMutex) lockBlocking(isReadLock bool) {
 				copy(dm.writeLocks, locks[:])
 			}
 
-			return
+			return true
+		}
+		if time.Since(start) >= timeout { // Are we past the timeout?
```

Comment:
That is correct, the design is not "real-time" in the sense that it is not guaranteed that you will get control back exactly at the time when the timeout expires.
It will be (short) time afterwards which depends on how the retry channel is configured for back-offs etc. 
So for our case this is sufficient precision as typically the time out will occur within 1 second after the expiry of the time out value.